models/index/writer.go

```diff
@@ -31,5 +31,5 @@ type Writer interface {
 	Payloads(height uint64, paths []ledger.Path, values []*ledger.Payload) error
 
 	Collections(height uint64, collections []*flow.LightCollection) error
-	Transactions(height uint64, transactions []*flow.TransactionBody) error
+	Transactions(height uint64, transactions []*flow.TransactionBody, results []*flow.TransactionResult) error
```

Comment:
The thing is that results have nothing to do with height, they're just indexed by transaction ID, so that's why I thought I'd bundle them up together instead, to not break the consistency. Currently all of the indexes that are by ID are indexed by methods that already index the primary entity by height, so I thought that adding `Results(results []*flow.TransactionResult) error` among other methods would seem weird and it wouldn't be clear why it's the only exception.

But if you think it's better this way, I'll change it now 👍 I'll do it in a specific commit, in case we want to revert it if you agree with my point above :p

Reply:
I definitely think they should be indexed separately, as they are separate DB entries, and we want control over it. It's true that they are not indexed by block _directly_, but they are _indirectly_ through the transaction IDs.